ang/prospect/case-actions/services/unlink-contribution-case-action.service.js

```diff
@@ -0,0 +1,52 @@
+(function (angular) {
+  var module = angular.module('prospect');
+
+  module.service('UnlinkContributionCaseAction', UnlinkContributionCaseAction);
+
+  /**
+   * @param {object} ProspectConverted Prospect Converted Service
+   */
+  function UnlinkContributionCaseAction (ProspectConverted) {
+    /**
+     * Checks if the Action is allowed
+     *
+     * @param {object} action action
+     * @param {Array} cases cases
+     * @returns {boolean} if action is allowed
+     */
+    this.isActionAllowed = function (action, cases) {
+      if (!cases[0] || !cases[0].prospect.paymentInfo) {
+        return;
+      }
+
+      var isContributeTypeProspect =
+        cases[0].prospect.paymentInfo.payment_entity === 'contribute';
+
+      return cases[0] &&
```

Comment:
I think at this stage the `cases[0]` has no chance to be false, then it would be unnecessary to evaluate it.

Reply:
No it can be false, for case bulk actions. Before you select any cases, this array would be empty. So need to check.

Reply:
yes, but I mean, it was checked a few lines before, on this method

Reply:
Oh Okay, yep I removed that already :D (while fixing other comments), thanks. I thought you are talking about the first line.